core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/SubmissionCostService.scala

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import java.util
+import com.google.api.services.bigquery.model.TableRow
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.google.HttpGoogleBigQueryDAO
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+object SubmissionCostService {
+  def constructor(tableName: String, bigQueryDAO: HttpGoogleBigQueryDAO)(implicit executionContext: ExecutionContext) =
+    new SubmissionCostService(tableName, bigQueryDAO)
+}
+
+class SubmissionCostService(tableName: String, bigQueryDAO: HttpGoogleBigQueryDAO)(implicit val executionContext: ExecutionContext) {
+
+
+  def getWorkflowCosts(namespace: String,
+                       workflowIds: Seq[String],
+                       googleProject: GoogleProject): Future[Map[String, Float]] = {
+
+    extractWorkflowCostResults(executeWorkflowCostsQuery(namespace, workflowIds, googleProject))
+  }
+
+  /*
+   * Manipulates and massages a BigQuery result.
+   */
+  def extractWorkflowCostResults(rowsFuture: Future[util.List[TableRow]]): Future[Map[String, Float]] = {
+
+    rowsFuture map { rowsOrNull =>
+      Option(rowsOrNull) match {
+        case Some(rows) => rows.asScala.map { row =>
+          // workflow ID is contained in the 2nd cell, cost is contained in the 3rd cell
+          row.getF.get(1).getV.toString -> row.getF.get(2).getV.toString.toFloat
+        }.toMap
+        case None => Map.empty[String, Float]
+      }
+    }
+  }
+
+  /*
+   * Queries BigQuery for compute costs associated with the workflowIds.
+   */
+  private def executeWorkflowCostsQuery(namespace: String,
+                       workflowIds: Seq[String],
+                       googleProject: GoogleProject): Future[util.List[TableRow]] = {
+
+    // TODO: need to protect against SQL injection. Can we use Slick? Hate to bring in Slick just for that.
```

Comment:
we do need to resolve the injection issue - let's talk in person.

Reply:
I have a way! But need some assistance with sbt dependencies.